### PR TITLE
Shell scripts alignment

### PR DESCRIPTION
**Pull request by gbe**
This PR is to improve the general quality of shell scripts:
- Local variable names are switched to lowercase,
- Some local variables were not declared as local,
- Double brackets were not used systematically in if conditions.

Fingers crossed nothing is broken.

### CODE REVIEW

File: sources/install/package_ad.sh [hunk 1]
```diff
@@ -111,8 +112,8 @@ function install_bloodhound-py() {
     colorecho "Installing and Python ingestor for BloodHound"
     pipx install git+https://github.com/fox-it/BloodHound.py
     # https://github.com/dirkjanm/BloodHound.py/pull/146
-    local TEMP_FIX_LIMIT="2024-02-15"
-    if [ "$(date +%Y%m%d)" -gt "$(date -d $TEMP_FIX_LIMIT +%Y%m%d)" ]; then
+    local temp_fix_limit="2024-02-15"
```
Comment (ShutdownRepo) [inline sources/install/package_ad.sh:115]: fyi, this needs to be changed as well in the doc ([here](https://exegol.readthedocs.io/en/latest/community/contributors.html#temporary-fixing-a-tool)). I did the change now

**Review by ShutdownRepo — APPROVED** [review]

File: sources/install/package_ad.sh [hunk 2]
```diff
@@ -97,7 +98,7 @@ function install_crackmapexec() {
     git -C /opt/tools/ clone --depth 1 https://github.com/Porchetta-Industries/CrackMapExec
     pipx install /opt/tools/CrackMapExec/
     mkdir -p ~/.cme
-    [ -f ~/.cme/cme.conf ] && mv ~/.cme/cme.conf ~/.cme/cme.conf.bak
+    [[ -f "~/.cme/cme.conf" ]] && mv ~/.cme/cme.conf ~/.cme/cme.conf.bak
```
Comment (ShutdownRepo) [inline sources/install/package_ad.sh:101]: Tilde doesn't expand in quotes. Quotes need to be removed.

File: sources/install/package_ad.sh [hunk 3]
```diff
@@ -1141,7 +1143,7 @@ function install_netexec() {
     git -C /opt/tools/ clone --depth 1 https://github.com/Pennyw0rth/NetExec
     pipx install /opt/tools/NetExec/
     mkdir -p ~/.nxc
-    [ -f ~/.nxc/nxc.conf ] && mv ~/.nxc/nxc.conf ~/.nxc/nxc.conf.bak
+    [[ -f "~/.nxc/nxc.conf" ]] && mv ~/.nxc/nxc.conf ~/.nxc/nxc.conf.bak
```
Comment (ShutdownRepo) [inline sources/install/package_ad.sh:1146]: Tilde doesn't expand in quotes. Quotes need to be removed.

File: sources/assets/exegol/start.sh [hunk 4]
```diff
@@ -2,38 +2,38 @@
...
-    COMPRESS=$3
+    local compress=$3
 
-    # Logging shell using $METHOD and spawn a $USER_SHELL shell
+    # Logging shell using $method and spawn a $user_shell shell
 
     umask 007
     mkdir -p /workspace/logs/
-    FILELOG="/workspace/logs/$(date +%d-%m-%Y_%H-%M-%S)_shell.${METHOD}"
+    local filelog="/workspace/logs/$(date +%d-%m-%Y_%H-%M-%S)_shell.${method}"
```
Comment (ShutdownRepo) [inline sources/assets/exegol/start.sh:15]: This needs to be declared and assigned separately, like so
```bash
local filelog
filelog="/workspace/logs/$(date +%d-%m-%Y_%H-%M-%S)_shell.${method}"
```

**Review by ShutdownRepo — CHANGES_REQUESTED** [review]
Shellcheck errors raised by the pipeline

File: sources/install/package_ad.sh (hunk 1 above)
Comment (gbe) [reply sources/install/package_ad.sh:115]: Thanks

File: sources/install/package_ad.sh (hunk 3 above)
Comment (gbe) [reply sources/install/package_ad.sh]: Done

File: sources/assets/exegol/start.sh (hunk 4 above)
Comment (gbe) [reply sources/assets/exegol/start.sh]: Done

File: sources/install/package_ad.sh (hunk 2 above)
Comment (gbe) [reply sources/install/package_ad.sh]: Done

**Comment by ShutdownRepo** [comment]
> Hello @ShutdownRepo,
> 
> Thank you for the review. Shame on me ;) All the points you raised shoud be fixed now.

I would've done the same mistakes, no big deal, that's why we have machines doing the checks ;)

**Pull request merged**